### DISCUSSION TITLE
Width of side panels

### DISCUSSION THREAD

**Discussion by mojoaar** [Budibase/budibase#9321] [Help] [answered]
Hi.

I'm trying to control the width of these new side panels, but cant seem to figure out which component I should edit to do it.

Right now it'll just use auto and set it to the widest field (I imagine at least). I have eject the blocks that was auto build from the data structure.

I hope somebody can give me a pointer to get more width on the side panels?

Side panel with highlight of more width:
![bb_sp](https://user-images.githubusercontent.com/21560785/211984743-2bb9a938-b8d6-45d7-91b4-e066970b6725.png)

Ejected blocks:
![bb_sp1](https://user-images.githubusercontent.com/21560785/211984788-c4e68a31-4a47-46e5-9e10-3a20bf895a9d.png)

**Comment by avivkri** [comment, marked as answer]
Hello @mojoaar,
I'm not sure if there's a simpler way. I've been using the Embed component with this content to set custom width for side panels. This Embed component has to be placed in every screen where the side panel needs custom width.

```
<style>
#side-panel-container {
  width: 500px !important;
}
</style>
```

**Reply by mojoaar** [reply]
@avivkri Thx for the suggestions - I was really hoping for a more "simple" solution in the design ui.

**Reply by melohagan** [reply]
Thanks @avivkri (I wonder if you can do it without the svelte IDs. I'm assuming they're randomly generated so might not work for other users).

We had an internal discussion around this, and we decided that it was best not to allow the user to increase the width of the panel. The reason is that the default width was perfect for the main purpose - forms.

Increasing the width looked a bit odd:
![image](https://user-images.githubusercontent.com/101575380/214025171-66f5837e-04d9-4087-b795-6a947fb057ba.png)

You can still create modals as before if you want the form to be more in focus.

**Reply by avivkri** [reply]
Thanks @melohagan, Good to know about the svelte IDs. It works without them too, but the !important rule override had to be added. Updated the answer so that it applies to all.

I agree the width does look odd in the screenshot with lot of white space on either side. It may be more jarring in wide screen monitors. In my case, I wanted to avoid some form elements getting wrapped in default width. Also, the Side panel feels more smoother than the Modal even when they both have the same number of fields.
